soh/soh/Enhancements/randomizer/3drando/spoiler_log.cpp

```diff
@@ -501,27 +501,46 @@ static void WriteMasterQuestDungeons(tinyxml2::XMLDocument& spoilerLog) {
   }
 }
 
-// Writes the required trails to the spoiler log, if there are any.
-static void WriteRequiredTrials(tinyxml2::XMLDocument& spoilerLog) {
-  auto parentNode = spoilerLog.NewElement("required-trials");
-
-  for (const auto* trial : Trial::trialList) {
-    if (trial->IsSkipped()) {
-      continue;
+// Writes the required trials to the spoiler log, if there are any.
+static void WriteRequiredTrials() {
+    for (const auto& trial : Trial::trialList) {
+        if (trial->IsRequired()) {
+            std::string trialName;
+            switch (gSaveContext.language) {
+                case LANGUAGE_FRA:
+                    trialName = trial->GetName().GetFrench();
+                    break;
+                case LANGUAGE_ENG:
+                default:
+                    trialName = trial->GetName().GetEnglish();
+                    break;
+            }
+            jsonData["requiredTrials"].push_back(RemoveLineBreaks(trialName));
+        }
     }
-
-    auto node = parentNode->InsertNewChildElement("trial");
-    // PURPLE TODO: LOCALIZATION
-    std::string name = trial->GetName().GetEnglish();
-    name[0] = toupper(name[0]); // Capitalize T in "The"
-    node->SetAttribute("name", name.c_str());
-  }
-
-  if (!parentNode->NoChildren()) {
-    spoilerLog.RootElement()->InsertEndChild(parentNode);
-  }
 }
 
+// Writes the required trails to the spoiler log, if there are any.
+// static void WriteRequiredTrials(tinyxml2::XMLDocument& spoilerLog) {
+//   auto parentNode = spoilerLog.NewElement("required-trials");
+
+//   for (const auto* trial : Trial::trialList) {
+//     if (trial->IsSkipped()) {
+//       continue;
+//     }
+
+//     auto node = parentNode->InsertNewChildElement("trial");
+//     // PURPLE TODO: LOCALIZATION
+//     std::string name = trial->GetName().GetEnglish();
+//     name[0] = toupper(name[0]); // Capitalize T in "The"
+//     node->SetAttribute("name", name.c_str());
+//   }
+
+//   if (!parentNode->NoChildren()) {
+//     spoilerLog.RootElement()->InsertEndChild(parentNode);
+//   }
+// }
+
```

Comment:
Leftover code from before it was done I'm guessing?

Reply:
3drando code

Reply:
Actually that was the original code before I wrote a new function.

Reply:
Then my follow up question is, do we want to keep it around?

Reply:
I did go ahead and remove it though.